### DISCUSSION TITLE
How to quarto_render into console environment

### DISCUSSION THREAD

**Discussion by erikerhardt** [quarto-dev/quarto-cli#1773] [Q&A] [answered]
I want quarto::quarto_render() to use the console environment for execution, as rmarkdown::render() does.  I have tried the debug=TRUE, execute_debug=TRUE, and as_job=FALSE/TRUE options.

After rendering a document, I want to inspect and use the objects in the console environment for further development.  I don't want to have to use the RStudio GUI to "run all" code chunks.

**Comment by jjallaire** [comment, marked as answer]
There isn't a direct path to the R environment from `quarto::quarto_render()` as its invoking an executable which in turn invokes R (this differs from `rmarkdown::render()` which does everything in R natively). 

You can however call the `knitr::purl()` function on your qmd to extract all of the code into a .R script and proceed from there.

**Reply by andrewpbray** [reply]
Is it possible to extract hash pipe cell options from a .qmd file using `knitr::purl()?`

It looks like knitr cell options in `{}` can be extracted with `knitr::purl("practice-quiz.qmd", documentation = 1)`, but not the options inside the cell behind a `#|`. I figure this may be something on the to-do for `knitr` I'm wondering if this is something that quarto would ever do seeing as how the sort of parsing done by `knitr::purl` is all pre-compute, as I understand it. Or perhaps this is a task best left to a filter extension?

**Reply by jjallaire** [reply]
There isn't currently a way to do this. @cderv or @yihui do you know if this is planned?

**Reply by cderv** [reply]
This was not on our radar yet, but I am sure we can find a way to make purl support in-chunk options. 

I opened https://github.com/yihui/knitr/issues/2165 to track this and discuss.

**Reply by cderv** [reply]
@andrewpbray dev version of **knitr** should now support extracting hash pipe cell options  commented lines.
